builtin/docker/builder.go

```diff
@@ -213,11 +222,18 @@ func (b *Builder) Build(
 		// We set this to true so we use the img-based injector later
 		useImg = true
 	} else {
+		buildArgs := make(map[string]*string)
```

Comment:
I should of noticed this the first time I reviewed this PR, but why not make the config option `BuildArgs` a `map[string]*string` to begin with? I think this would save us a translation here. Sorry I didn't notice this the first time around!